### ISSUE TITLE
MeanDim Tests causing a crash for MHLO

### ISSUE THREAD

**Issue by vivekkhandelwal1** [llvm/torch-mlir#1626]
Hi, after the merging of this PR: https://github.com/llvm/torch-mlir/pull/1605, the MeanDim tests are causing a crash, with the following stack trace:
```
torch-mlir-opt: /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/lib/Dialect/TorchConversion/Transforms/BackendTypeConversion.cpp:41: auto setupValueTensorToBuiltinTensorConversion(mlir::ConversionTarget &, mlir::TypeConverter &)::(anonymous class)::operator()(mlir::OpBuilder &, mlir::TensorType, mlir::ValueRange, mlir::Location) const: Assertion `inputs[0].getType().isa<Torch::BaseTensorType>()' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/build/bin/torch-mlir-opt mhlo_test.mlir -torch-backend-to-mhlo-backend-pipeline
 #0 0x00000000077e2dda llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/llvm/lib/Support/Unix/Signals.inc:569:11
 #1 0x00000000077e2fab PrintStackTraceSignalHandler(void*) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/llvm/lib/Support/Unix/Signals.inc:636:1
 #2 0x00000000077e151b llvm::sys::RunSignalHandlers() /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/llvm/lib/Support/Signals.cpp:103:5
 #3 0x00000000077e3721 SignalHandler(int) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/llvm/lib/Support/Unix/Signals.inc:407:1
 #4 0x00007f41a76b0420 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x14420)
 #5 0x00007f41a709a00b raise /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/raise.c:51:1
 #6 0x00007f41a7079859 abort /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:81:7
 #7 0x00007f41a7079729 get_sysdep_segment_value /build/glibc-SzIz7B/glibc-2.31/intl/loadmsgcat.c:509:8
 #8 0x00007f41a7079729 _nl_load_domain /build/glibc-SzIz7B/glibc-2.31/intl/loadmsgcat.c:970:34
 #9 0x00007f41a708afd6 (/lib/x86_64-linux-gnu/libc.so.6+0x33fd6)
#10 0x0000000005a5d8ba setupValueTensorToBuiltinTensorConversion(mlir::ConversionTarget&, mlir::TypeConverter&)::$_1::operator()(mlir::OpBuilder&, mlir::TensorType, mlir::ValueRange, mlir::Location) const /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/lib/Dialect/TorchConversion/Transforms/BackendTypeConversion.cpp:0:5
#11 0x0000000005a5d731 std::function<llvm::Optional<mlir::Value> (mlir::OpBuilder&, mlir::Type, mlir::ValueRange, mlir::Location)> mlir::TypeConverter::wrapMaterialization<mlir::TensorType, setupValueTensorToBuiltinTensorConversion(mlir::ConversionTarget&, mlir::TypeConverter&)::$_1>(setupValueTensorToBuiltinTensorConversion(mlir::ConversionTarget&, mlir::TypeConverter&)::$_1&&)::'lambda'(mlir::OpBuilder&, mlir::Type, mlir::ValueRange, mlir::Location)::operator()(mlir::OpBuilder&, mlir::Type, mlir::ValueRange, mlir::Location) const /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/llvm/../mlir/include/mlir/Transforms/DialectConversion.h:308:16
#12 0x0000000005a5d573 std::_Function_handler<llvm::Optional<mlir::Value> (mlir::OpBuilder&, mlir::Type, mlir::ValueRange, mlir::Location), std::function<llvm::Optional<mlir::Value> (mlir::OpBuilder&, mlir::Type, mlir::ValueRange, mlir::Location)> mlir::TypeConverter::wrapMaterialization<mlir::TensorType, setupValueTensorToBuiltinTensorConversion(mlir::ConversionTarget&, mlir::TypeConverter&)::$_1>(setupValueTensorToBuiltinTensorConversion(mlir::ConversionTarget&, mlir::TypeConverter&)::$_1&&)::'lambda'(mlir::OpBuilder&, mlir::Type, mlir::ValueRange, mlir::Location)>::_M_invoke(std::_Any_data const&, mlir::OpBuilder&, mlir::Type&&, mlir::ValueRange&&, mlir::Location&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/std_function.h:285:9
#13 0x00000000072e210a std::function<llvm::Optional<mlir::Value> (mlir::OpBuilder&, mlir::Type, mlir::ValueRange, mlir::Location)>::operator()(mlir::OpBuilder&, mlir::Type, mlir::ValueRange, mlir::Location) const /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/std_function.h:688:14
#14 0x00000000072c1f59 mlir::TypeConverter::materializeConversion(llvm::MutableArrayRef<std::function<llvm::Optional<mlir::Value> (mlir::OpBuilder&, mlir::Type, mlir::ValueRange, mlir::Location)> >, mlir::OpBuilder&, mlir::Location, mlir::Type, mlir::ValueRange) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Transforms/Utils/DialectConversion.cpp:3042:34
#15 0x00000000057299d2 mlir::TypeConverter::materializeTargetConversion(mlir::OpBuilder&, mlir::Location, mlir::Type, mlir::ValueRange) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/include/mlir/Transforms/DialectConversion.h:224:12
#16 0x00000000072d12df legalizeUnresolvedMaterialization((anonymous namespace)::UnresolvedMaterialization&, llvm::DenseMap<mlir::Operation*, (anonymous namespace)::UnresolvedMaterialization*, llvm::DenseMapInfo<mlir::Operation*, void>, llvm::detail::DenseMapPair<mlir::Operation*, (anonymous namespace)::UnresolvedMaterialization*> >&, mlir::ConversionPatternRewriter&, mlir::detail::ConversionPatternRewriterImpl&, llvm::DenseMap<mlir::Value, llvm::SmallVector<mlir::Value, 6u>, llvm::DenseMapInfo<mlir::Value, void>, llvm::detail::DenseMapPair<mlir::Value, llvm::SmallVector<mlir::Value, 6u> > >&) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Transforms/Utils/DialectConversion.cpp:2783:39
#17 0x00000000072cfc98 (anonymous namespace)::OperationConverter::legalizeUnresolvedMaterializations(mlir::ConversionPatternRewriter&, mlir::detail::ConversionPatternRewriterImpl&, llvm::Optional<llvm::DenseMap<mlir::Value, llvm::SmallVector<mlir::Value, 6u>, llvm::DenseMapInfo<mlir::Value, void>, llvm::detail::DenseMapPair<mlir::Value, llvm::SmallVector<mlir::Value, 6u> > > >&) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Transforms/Utils/DialectConversion.cpp:2823:16
#18 0x00000000072caf5c (anonymous namespace)::OperationConverter::finalize(mlir::ConversionPatternRewriter&) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Transforms/Utils/DialectConversion.cpp:2484:14
#19 0x00000000072c396a (anonymous namespace)::OperationConverter::convertOperations(llvm::ArrayRef<mlir::Operation*>, llvm::function_ref<void (mlir::Diagnostic&)>) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Transforms/Utils/DialectConversion.cpp:2460:14
#20 0x00000000072c3599 mlir::applyPartialConversion(llvm::ArrayRef<mlir::Operation*>, mlir::ConversionTarget&, mlir::FrozenRewritePatternSet const&, llvm::DenseSet<mlir::Operation*, llvm::DenseMapInfo<mlir::Operation*, void> >*) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Transforms/Utils/DialectConversion.cpp:3359:22
#21 0x00000000072c3b42 mlir::applyPartialConversion(mlir::Operation*, mlir::ConversionTarget&, mlir::FrozenRewritePatternSet const&, llvm::DenseSet<mlir::Operation*, llvm::DenseMapInfo<mlir::Operation*, void> >*) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Transforms/Utils/DialectConversion.cpp:3365:10
#22 0x0000000005a6eb5a (anonymous namespace)::ConvertTorchToMhlo::runOnOperation() /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:75:16
#23 0x000000000737a950 mlir::detail::OpToOpPassAdaptor::run(mlir::Pass*, mlir::Operation*, mlir::AnalysisManager, bool, unsigned int) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Pass/Pass.cpp:466:21
#24 0x000000000737af5a mlir::detail::OpToOpPassAdaptor::runPipeline(mlir::OpPassManager&, mlir::Operation*, mlir::AnalysisManager, bool, unsigned int, mlir::PassInstrumentor*, mlir::PassInstrumentation::PipelineParentInfo const*) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Pass/Pass.cpp:529:16
#25 0x0000000007380766 mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::$_14::operator()(mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo&) const /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Pass/Pass.cpp:749:36
#26 0x00000000073803ac mlir::LogicalResult mlir::failableParallelForEach<__gnu_cxx::__normal_iterator<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo*, std::vector<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo, std::allocator<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo> > >, mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::$_14&>(mlir::MLIRContext*, __gnu_cxx::__normal_iterator<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo*, std::vector<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo, std::allocator<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo> > >, __gnu_cxx::__normal_iterator<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo*, std::vector<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo, std::allocator<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo> > >, mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::$_14&) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/include/mlir/IR/Threading.h:46:18
#27 0x000000000737c2a6 mlir::LogicalResult mlir::failableParallelForEach<std::vector<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo, std::allocator<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo> >&, mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::$_14&>(mlir::MLIRContext*, std::vector<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo, std::allocator<mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::OpPMInfo> >&, mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool)::$_14&) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/include/mlir/IR/Threading.h:92:10
#28 0x000000000737bb26 mlir::detail::OpToOpPassAdaptor::runOnOperationAsyncImpl(bool) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Pass/Pass.cpp:759:14
#29 0x000000000737ac0d mlir::detail::OpToOpPassAdaptor::runOnOperation(bool) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Pass/Pass.cpp:650:5
#30 0x000000000737a93e mlir::detail::OpToOpPassAdaptor::run(mlir::Pass*, mlir::Operation*, mlir::AnalysisManager, bool, unsigned int) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Pass/Pass.cpp:463:5
#31 0x000000000737af5a mlir::detail::OpToOpPassAdaptor::runPipeline(mlir::OpPassManager&, mlir::Operation*, mlir::AnalysisManager, bool, unsigned int, mlir::PassInstrumentor*, mlir::PassInstrumentation::PipelineParentInfo const*) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Pass/Pass.cpp:529:16
#32 0x000000000737c9d3 mlir::PassManager::runPasses(mlir::Operation*, mlir::AnalysisManager) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Pass/Pass.cpp:834:10
#33 0x000000000737c8ea mlir::PassManager::run(mlir::Operation*) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Pass/Pass.cpp:814:60
#34 0x000000000293de4a performActions(llvm::raw_ostream&, bool, bool, llvm::SourceMgr&, mlir::MLIRContext*, llvm::function_ref<mlir::LogicalResult (mlir::PassManager&)>, bool, bool) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:91:17
#35 0x000000000293daa3 processBuffer(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, bool, bool, bool, bool, bool, bool, llvm::function_ref<mlir::LogicalResult (mlir::PassManager&)>, mlir::DialectRegistry&, llvm::ThreadPool*) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:139:12
#36 0x000000000293d846 mlir::MlirOptMain(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, llvm::function_ref<mlir::LogicalResult (mlir::PassManager&)>, mlir::DialectRegistry&, bool, bool, bool, bool, bool, bool, bool)::$_0::operator()(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, llvm::raw_ostream&) const /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:181:12
#37 0x000000000293d725 mlir::LogicalResult llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, llvm::raw_ostream&)>::callback_fn<mlir::MlirOptMain(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, llvm::function_ref<mlir::LogicalResult (mlir::PassManager&)>, mlir::DialectRegistry&, bool, bool, bool, bool, bool, bool, bool)::$_0>(long, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, llvm::raw_ostream&) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/llvm/include/llvm/ADT/STLFunctionalExtras.h:45:12
#38 0x0000000007653c91 llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, llvm::raw_ostream&)>::operator()(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, llvm::raw_ostream&) const /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/llvm/include/llvm/ADT/STLFunctionalExtras.h:68:12
#39 0x0000000007653211 mlir::splitAndProcessBuffer(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, llvm::raw_ostream&)>, llvm::raw_ostream&, bool, bool) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Support/ToolUtilities.cpp:28:12
#40 0x000000000293c67e mlir::MlirOptMain(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, llvm::function_ref<mlir::LogicalResult (mlir::PassManager&)>, mlir::DialectRegistry&, bool, bool, bool, bool, bool, bool, bool) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:186:10
#41 0x000000000293c81e mlir::MlirOptMain(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >, mlir::PassPipelineCLParser const&, mlir::DialectRegistry&, bool, bool, bool, bool, bool, bool, bool, bool) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:209:10
#42 0x000000000293d531 mlir::MlirOptMain(int, char**, llvm::StringRef, mlir::DialectRegistry&, bool) /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/externals/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:306:14
#43 0x00000000028f6ddd main /home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/tools/torch-mlir-opt/torch-mlir-opt.cpp:26:7
#44 0x00007f41a707b083 __libc_start_main /build/glibc-SzIz7B/glibc-2.31/csu/../csu/libc-start.c:342:3
#45 0x00000000028f6c9e _start (/home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/build/bin/torch-mlir-opt+0x28f6c9e)
Aborted (core dumped)
```

**Comment by silvasean** [comment]
https://github.com/llvm/torch-mlir/pull/1605 should have included an appropriate xfail. We should revert it or fix this forward.

You will need to use ``crashing_tests_to_not_attempt_to_run_and_a_bug_is_filed` since it is a crash (we cannot xfail it) https://github.com/llvm/torch-mlir/blob/68f568b7041dd216a0d34901fcee7375e57840c6/e2e_testing/main.py#L65

**Comment by tanyokwok** [comment]
> Hi @tanyokwok, till the fix is added could you please add them to the xfail set? Since all the PRs are stalled on this.

I found it's because the input and output types of aten::sum mismatch. I will fix it in https://github.com/llvm/torch-mlir/pull/1634

**Comment by vivekkhandelwal1** [comment]
> > Hi @tanyokwok, till the fix is added could you please add them to the xfail set? Since all the PRs are stalled on this.
> 
> I found it's because the input and output types of aten::sum mismatch. I will fix it in #1634

Thanks @tanyokwok!
